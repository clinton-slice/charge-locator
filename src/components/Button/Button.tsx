import {
  StyleSheet,
  Text,
  TouchableOpacity,
  ViewStyle,
  StyleProp,
} from "react-native";

interface Props {
  onPress: () => void;
  style?: StyleProp<ViewStyle>;
  label: string;
}

export const Button = ({ onPress, style, label }: Props) => {
  return (
    <TouchableOpacity style={[styles.button, style]} onPress={onPress}>
      <Text style={styles.text}>{label}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    padding: 16,
    fontSize: 14,
    flex: 1,
    alignItems: "center",
  },
  navButton: {
    borderRightWidth: 1,
    borderRightColor: "#bfbfbf",
  },
  text: {
    fontSize: 14,
    fontWeight: "bold",
  },
});
