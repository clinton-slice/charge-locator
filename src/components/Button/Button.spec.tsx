import React from "react";
import { render, screen, fireEvent } from "@testing-library/react-native";
import Button from ".";

describe("Button", () => {
  const mockOnPress = jest.fn();
  const component = <Button onPress={mockOnPress} label="Button" />;
  it("renders the correct message", () => {
    const { container } = render(component);
    expect(container.firstChild).toMatchSnapshot();
  });
  it("calls `mockOnPress` callback when clicked", () => {
    render(component);
    fireEvent.press(screen.getByText("Button"));
    expect(mockOnPress).toHaveBeenCalled();
  });
});
