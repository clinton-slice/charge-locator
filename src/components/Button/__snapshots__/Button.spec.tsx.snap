// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders the correct message 1`] = `undefined`;
